File: Scripts/src/icarus/model/EL/container/formelement/formtextarea/FORMTEXTAREA.js
/** @module */
import FORMELEMENT, { ATTRIBUTES, EL, LABEL, MODEL } from '../FORMELEMENT.js';
import TEXTAREA from '../../../textarea/TEXTAREA.js';
/** Represents a <TEXTAREA> for an Icarus Form       
    @class
    @extends FORMELEMENT
*/
export default class FORMTEXTAREA extends FORMELEMENT {
    constructElements() {
        return this.chain(() => {
            this.input = new TEXTAREA(this.body.pane, new MODEL(new ATTRIBUTES({
                name: this.attributes.name
            })), this.attributes.value || '');
        });
    }
}
export { ATTRIBUTES, EL, LABEL, MODEL, TEXTAREA }

File: Models/Icarus/Form/FORMTEXTAREA.cs
using ICARUS.Models.Icarus;
using ICARUS.Models.Icarus.Elements;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Web.Mvc;
namespace ICARUS.Models {
    /// <summary>
    /// A Form Text Area
    /// </summary>
    public class FORMTEXTAREA : Container {
        /// <summary>
        /// Generic FORMTEXTAREA Constructor for Entity
        /// </summary>
        public FORMTEXTAREA() : base("DIV", new MODEL() {
            label = "FORMTEXTAREA"
        }) {

        }
        /// <summary>
        /// Constructs an FORMTEXTAREA with the given id and label attributes already set
        /// </summary>
        /// <param name="formPost"></param>
        public FORMTEXTAREA(FormPost formPost) : base("FORMTEXTAREA", formPost) {

        }
    }
}